connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -96,6 +99,7 @@ public class MirrorSourceConnector extends SourceConnector {
     private Admin sourceAdminClient;
     private Admin targetAdminClient;
     private Admin offsetSyncsAdminClient;
+    private volatile String useIncrementalAlterConfigs;
```

Comment:
I think it may be a bit simpler to replace this with a boolean value. Most of the time, we're only concerned about which API (incremental or deprecated) that we use. The only time the distinction between requested and required comes into play is if we see an `UnsupportedVersionException` when trying to use the incremental API, and at that point, we can just use `config::useIncrementalAlterConfigs` to check the value and then (if necessary) update the value of the `useIncrementalAlterConfigs` boolean instance field.